packages/data/src/world/mq/gerudo_training_grounds_mq.yml

```diff
@@ -23,40 +19,171 @@
-    "MQ Gerudo Training Grounds SR Lava Front": "can_use_bow || (can_longshot && has_fire) || (can_hookshot && trick(OOT_MQ_GTG_FLAMES)) || (has_hover_boots && trick(OOT_MQ_GTG_FLAMES) && has_fire_or_sticks)"
-    "MQ Gerudo Training Grounds Maze Right Side Middle Chest": "can_use_bow && can_hookshot && event(GTG_LAVA_HAMMER)"
-    "MQ Gerudo Training Grounds Maze Right Side Right Chest": "can_use_bow && can_hookshot && event(GTG_LAVA_HAMMER)"
+    GTG_LAVA_HOOK_TARGETS: "silver_rupees_gtg_lava_mq"
+    GTG_MQ_LAVA_TORCH_ENTRANCE_SIDE: "can_use_bow"
+    GTG_MQ_LAVA_TORCH_FAR: "has_fire_arrows"
+    GTG_MQ_LAVA_TORCH_COMMON_PLATFORMS_ENTRANCE: "(event(GTG_MQ_LAVA_TORCH_ENTRANCE_SIDE) && (is_adult || (is_child && (has_hover_boots || climb_anywhere || ((event(GTG_LAVA_HOOK_TARGETS) && can_hookshot) || hookshot_anywhere))))) || (trick(OOT_MQ_GTG_FLAMES) && (has_hover_boots || (can_hookshot && event(GTG_LAVA_HOOK_TARGETS)) || hookshot_anywhere))"
+    #GTG_MQ_LAVA_SR_HOOK_ANYWHERE: "settings.silverRupeeShuffle && hookshot_anywhere"
+  locations:
+    "MQ Gerudo Training Grounds SR Lava Back-Left": "event(GTG_MQ_LAVA_TORCH_COMMON_PLATFORMS_ENTRANCE)"
+    "MQ Gerudo Training Grounds SR Lava Back-Right": "event(GTG_MQ_LAVA_TORCH_COMMON_PLATFORMS_ENTRANCE)"
+    "MQ Gerudo Training Grounds SR Lava Center": "event(GTG_MQ_LAVA_TORCH_COMMON_PLATFORMS_ENTRANCE)"
+    "MQ Gerudo Training Grounds SR Lava Front-Right": "event(GTG_MQ_LAVA_TORCH_COMMON_PLATFORMS_ENTRANCE)"
+    "MQ Gerudo Training Grounds SR Lava Front-Left": "event(GTG_MQ_LAVA_TORCH_COMMON_PLATFORMS_ENTRANCE)"
+    "MQ Gerudo Training Grounds SR Lava Front": "event(GTG_MQ_LAVA_TORCH_COMMON_PLATFORMS_ENTRANCE)"
+"Gerudo Training Grounds Lava Room Ledge Before Water Room":
+  dungeon: GTG
+  exits:
+    "Gerudo Training Grounds Lava Room Near Beginning": "can_longshot || event(GTG_MQ_LAVA_TORCH_WATER_ROOM_SIDE) || (trick(OOT_MQ_GTG_FLAMES) && (has_hover_boots || can_hookshot)) || hookshot_anywhere || climb_anywhere"
+    "Gerudo Training Grounds Lava Room Maze Ledge": "(event(GTG_MQ_LAVA_HAMMER_SWITCH) && can_longshot) || hookshot_anywhere || (climb_anywhere && event(GTG_LAVA_HOOK_TARGETS) && event(GTG_MQ_LAVA_TORCH_WATER_ROOM_SIDE)) || (event(GTG_MQ_LAVA_HAMMER_SWITCH) && can_hookshot && event(GTG_MQ_LAVA_TORCH_WATER_ROOM_SIDE))"
+    "Gerudo Training Grounds Lava Room Deep Side Ledge": "hookshot_anywhere || climb_anywhere"
+    "Gerudo Training Grounds Water Room": "silver_rupees_gtg_lava_mq"
+  events:
+    GTG_LAVA_HOOK_TARGETS: "silver_rupees_gtg_lava_mq"
+    GTG_MQ_LAVA_TORCH_WATER_ROOM_SIDE: "has_fire"
+    GTG_MQ_LAVA_TORCH_COMMON_PLATFORMS_WATER_SIDE: "(event(GTG_MQ_LAVA_TORCH_WATER_ROOM_SIDE) && (is_adult || (is_child && (has_hover_boots || climb_anywhere || ((event(GTG_LAVA_HOOK_TARGETS) && can_hookshot) || hookshot_anywhere))))) || (trick(OOT_MQ_GTG_FLAMES) && (has_hover_boots || (can_hookshot && event(GTG_LAVA_HOOK_TARGETS)) || hookshot_anywhere))"
```

Comment:
If you have an "is_age" alone as a requirement, you dont need to check for the other age for other requirements.

Reply:
Yes you do only because of jumping distance.

Reply:
No, you don't. If you are adult, it is satisfied by just seeing the "is_adult". Anything else implies you are not adult.

Reply:
Fixed.